tests/functional/adapter/materialization/test_on_table_exists.py

```diff
@@ -25,6 +25,94 @@ def models(self):
         }
 
 
+class TestOnTableExistsRename(BaseOnTableExists):
+    """
+    Testing on_table_exists = `rename` configuration for table materialization,
+    using dbt seed, run and tests commands and validate data load correctness.
+    """
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "table_rename",
+            "models": {"+materialized": "table", "+on_table_exists": "rename"},
+            "seeds": {
+                "+column_types": {"some_date": "timestamp(6)"},
+            },
+        }
+
+    # The actual sequence of dbt commands and assertions
+    # pytest will take care of all "setup" + "teardown"
+    def test_run_seed_test(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+        # run models two times to check on_table_exists = 'rename'
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+        # test tests
+        results = run_dbt(["test"], expect_pass=True)
+        assert len(results) == 3
+
+        # check if the data was loaded correctly
+        check_relations_equal(project.adapter, ["seed", "materialization"])
+
+
+@pytest.mark.iceberg
+class TestOnTableExistsRenameIceberg(TestOnTableExistsRename):
+    pass
+
+
+class TestOnTableExistsRenameIncrementalFullRefresh(BaseOnTableExists):
+    """
+    Testing on_table_exists = `rename` configuration for incremental materialization and full refresh flag,
+    using dbt seed, run and tests commands and validate data load correctness.
+    """
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "table_rename",
+            "models": {"+materialized": "incremental", "+on_table_exists": "rename"},
+            "seeds": {
+                "+column_types": {"some_date": "timestamp(6)"},
+            },
+        }
+
+    # The actual sequence of dbt commands and assertions
+    # pytest will take care of all "setup" + "teardown"
+    def test_run_seed_test(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+        # run models two times to check on_table_exists = 'rename'
+        results, logs = run_dbt_and_capture(["--debug", "run"], expect_pass=True)
+        assert len(results) == 1
+        assert (
+            f'drop table if exists "{project.database}"."{project.test_schema}"."materialization"'
+            not in logs
+        )
+        assert "dbt_tmp" and "dbt_backup" not in logs
+        results, logs = run_dbt_and_capture(["--debug", "run", "--full-refresh"], expect_pass=True)
+        assert len(results) == 1
+        assert "dbt_tmp" and "dbt_backup" in logs
```

Comment:
```suggestion
        assert f'create table "{project.database}"."{project.test_schema}"."materialization"' in logs
        assert 'alter table' not in logs
        results, logs = run_dbt_and_capture(["--debug", "run"], expect_pass=True)
        assert len(results) == 1
        assert f'create table "{project.database}"."{project.test_schema}"."materialization__dbt_tmp"' in logs
        assert f'alter table "{project.database}"."{project.test_schema}"."materialization" rename to "{project.database}"."{project.test_schema}"."materialization__dbt_backup"' in logs
        assert f'alter table "{project.database}"."{project.test_schema}"."materialization__dbt_tmp" rename to "{project.database}"."{project.test_schema}"."materialization"' in logs
        assert f'drop table if exists "{project.database}"."{project.test_schema}"."materialization__dbt_backup"' in logs

```

Let's make these assertions more explicit.